tests/Admin/AdminTest.php

```diff
@@ -713,24 +713,39 @@ public function testSubClass()
         $admin->setSubject(new BlogPost());
         $this->assertSame(BlogPost::class, $admin->getClass());
 
-        $admin->setSubClasses(['extended1' => 'NewsBundle\Entity\PostExtended1', 'extended2' => 'NewsBundle\Entity\PostExtended2']);
+        $admin->setSubClasses([
+            'extended1' => 'NewsBundle\Entity\PostExtended1',
+            'extended2' => 'NewsBundle\Entity\PostExtended2'
+        ]);
         $this->assertFalse($admin->hasSubClass('test'));
         $this->assertTrue($admin->hasSubClass('extended1'));
         $this->assertFalse($admin->hasActiveSubClass());
         $this->assertCount(2, $admin->getSubClasses());
         $this->assertNull($admin->getActiveSubClass());
         $this->assertNull($admin->getActiveSubclassCode());
-        $this->assertSame(BlogPost::class, $admin->getClass());
+        $this->assertSame(
+            BlogPost::class,
+            $admin->getClass(),
+            'When there is no subclass in the query the class parameter should be returned'
+        );
 
         $request = new Request(['subclass' => 'extended1']);
         $admin->setRequest($request);
         $this->assertFalse($admin->hasSubClass('test'));
         $this->assertTrue($admin->hasSubClass('extended1'));
         $this->assertTrue($admin->hasActiveSubClass());
         $this->assertCount(2, $admin->getSubClasses());
-        $this->assertSame(BlogPost::class, $admin->getActiveSubClass());
+        $this->assertSame(
+            'NewsBundle\Entity\PostExtended1',
```

Comment:
Do not introduce strings as name class, use ::class notation instead

Reply:
This class does not exist, it is a fake string injected into the admin.

Reply:
@ElectricMaxxx it does not need to exist: https://twitter.com/greg0ire/status/936958997151797250

Reply:
but i would not be able to colapse them throug a use statement.

Reply:
What do you mean by collapse?

Reply:
Collapse => mean import by a use statement. 

Is that impotant? At the end this is a unit test, which inject a hash of strings an we check if get a string back.

Reply:
Not important, no

Reply:
So what is missing now?

Reply:
Squashing your commits together and using this commit message: https://github.com/sonata-project/SonataAdminBundle/pull/4824#issuecomment-350191319